reinsurancecontract.py

```diff
@@ -63,7 +63,19 @@ def mature(self, time):
         else: #TODO: ? Instead: if self.insurancetype == "proportional":
             self.contract.unreinsure()
 
-        if np.random.uniform(0,1,1) < 0.95:
+        if np.random.uniform(0,1,1) < 1:
             reinrisk = self.property_holder.create_reinrisk(time,self.category)
-            if reinrisk is not None and hasattr(self.insurer, 'reinrisks_kept'):
-                self.insurer.reinrisks_kept.append(reinrisk)
\ No newline at end of file
+
+            if reinrisk is not None:
+                new_nonproportional_risks = [reinrisk]
+                [reinrisks_per_categ, number_reinrisks_categ] = self.insurer.risks_reinrisks_organizer(new_nonproportional_risks)  #Here the new reinrisks are organized by category.
+
+                for repetition in range(self.insurer.recursion_limit):     # TODO: find an efficient way to stop the recursion if there are no more risks to accept or if it is not accepting any more over several iterations.
```

Comment:
I do not understand this repetition loop. (Please use more comments.)
But I gather that the idea is to clear the market in every time step. I may be wrong about this, but I believe this is going to be horribly time consuming, so that it would increase the simulation run time by orders of magnitude. Consequently, I am not a big fan of this approach.

Reply:
No it's not market clearing. It only tries to reinsure that contract (if it can). I reuse the code that we have in metainsuranceorg.py.